server/chat-commands.js

```diff
@@ -3796,6 +3796,60 @@ const commands = {
 	},
 	importinputloghelp: [`/importinputlog [inputlog] - Starts a battle with a given inputlog. Requires: + % @ & ~`],
 
+	rejectdraw: 'offertie',
+	rejecttie: 'offertie',
+	acceptdraw: 'offertie',
+	accepttie: 'offertie',
+	offerdraw: 'offertie',
+	offertie(target, room, user, connection, cmd) {
+		const battle = room.battle;
+		if (!battle) return this.errorReply("Must be in a battle room.");
+		if (!Config.allowrequestingties) {
+			return this.errorReply("This server does not allow offering ties.");
+		}
+		if (!this.can('roomvoice', null, room)) return;
+		if ((cmd === 'accepttie' || cmd === 'rejecttie') && !battle.players.some(player => player.wantsTie)) {
+			return this.errorReply("No other player is requesting a tie right now. It was probably canceled.");
+		}
+		const player = battle.playerTable[user.userid];
+		if (cmd === 'rejecttie' && battle.players.some(player => player.wantsTie)) {
+			if (!player) {
+				return this.errorReply("Must be a player to reject ties.");
+			}
+			if (player.wantsTie) player.wantsTie = false;
+			for (const player of battle.players) {
+				player.sendRoom(Chat.html`|uhtmlchange|offertie|(Tie offer rejected)`);
+			}
+			return this.add(`${user.userid} rejects to tie this battle`);
+		}
+		if (!battle.players.some(player => player.wantsTie)) {
+			this.add(`${user.name} is offering a tie.`);
+			room.update();
+			for (const otherPlayer of battle.players) {
+				if (otherPlayer !== player) {
+					otherPlayer.sendRoom(Chat.html`|uhtml|offertie|<button class="button" name="send" value="/accepttie"><strong>accept tie</strong></button> <button class="button" name="send" value="/rejecttie">reject</button>`);
+				} else {
+					player.wantsTie = true;
+				}
+			}
+		} else {
+			if (!player) {
+				return this.errorReply("Must be a player to accept ties");
+			}
+			if (!player.wantsTie) {
+				player.wantsTie = true;
+			} else {
+				return this.errorReply("You have already agreed to tie this battle.");
+			}
+			this.add(`${user.userid} agrees to tie this battle.`);
+			if (battle.players.every(player => player.wantsTie)) {
+				this.add(`All the players in this battle (${Chat.toListString(battle.players.map(player => player.name))}) agree to tie this battle.`);
```

Comment:
```suggestion
				if (battle.players.length > 2) {
					this.add(`All players have agreed to tie.`);
				}
```

- no need to report in two-player battles, "[user] agrees to tie" is enough
- no need to list users; "[user] agrees to tie" is enough